v2/index_gen_test.go

```diff
@@ -56,7 +95,7 @@ func TestReadOrGenerateIndex(t *testing.T) {
 			carFile, err := os.Open(tt.carPath)
 			require.NoError(t, err)
 			t.Cleanup(func() { assert.NoError(t, carFile.Close()) })
-			got, err := ReadOrGenerateIndex(carFile)
+			got, err := ReadOrGenerateIndex(carFile, tt.readOpts...)
 			if tt.wantErr {
```

Comment:
Should there be an else case here that does `require.NoError(t, err)`?

Reply:
Does this not suffice?
```
if tt.wantErr {
  require.Error(t, err)
}
```

Reply:
Doh; ignore me; thank you @dirkmc will add 🍻

Reply:
I'm saying:
```
if tt.wantErr {
  require.Error(t, err)
} else {
  require.NoError(t, err)
}
```

So that it verifies that error is nil if we're _not_ expecting an error

Reply:
Fixed here: https://github.com/ipld/go-car/pull/186 Thank you for pointing this out 🍻